### PR TITLE
[SPARK-4362] [MLLIB] Make prediction probability available in NaiveBayesModel

### PR DESCRIPTION
**Pull request by srowen**
Add predictProbabilities to Naive Bayes, return class probabilities.

Continues https://github.com/apache/spark/pull/6761

### CODE REVIEW

**Comment by andrewor14** [comment]
Jenkins, slow test please

**Comment by acidghost** [comment]
Looks good to me. There's no check that they sum to 1 or that the prediction from probabilities is right, but I don't think is necessary

**Comment by srowen** [comment]
The prediction is already checked in the existing tests. I can add a check for sum to 1 though, yes.

**Comment by jkbradley** [comment]
I'll take a look

File: mllib/src/test/scala/org/apache/spark/mllib/classification/NaiveBayesSuite.scala [hunk 1]
```diff
@@ -154,6 +155,29 @@ class NaiveBayesSuite extends SparkFunSuite with MLlibTestSparkContext {
...
+  }
+
+  /**
+   * @param model Multinomial Naive Bayes model
+   * @param testData input to compute posterior probabilities for
+   * @return posterior class probabilities (in order of labels) for input
+   */
+  private def expectedMultinomialProbabilities(model: NaiveBayesModel, testData: Vector) = {
+    val piVector = new BDV(model.pi)
+    // model.labels is row-major; treat it as col-major representation of transpose, and transpose:
```
Comment (jkbradley) [inline mllib/src/test/scala/org/apache/spark/mllib/classification/NaiveBayesSuite.scala]: "labels" --> "theta"

**Comment by jkbradley** [comment]
That one item doesn't really matter; LGTM if you want to merge.
Thanks!

**Pull request closed without merging**